java/Oleksii-Riabov/src/com/zoolatech/lecture2/tasks/_2/Robot.java

```diff
@@ -0,0 +1,174 @@
+package com.zoolatech.lecture2.tasks._2;
+
+public class Robot {
+    private final Room room;
+    Direction direction;
+    private int width;
+    private int height;
+
+    public Robot() {
+        this.room = new Room(1, 1);
+        this.direction = Direction.SOUTH;
+        this.width = 0;
+        this.height = 0;
+    }
+
+    public Robot(int roomWidth, int roomHeight, String direction, int robotsStartingWidth, int robotStartingHeight) {
+        this.room = new Room(roomWidth, roomHeight);
+        this.direction = Direction.valueOf(direction);
+        this.width = robotsStartingWidth;
+        this.height = robotStartingHeight;
+    }
+
+    public Room getRoom() {
+        return room;
+    }
+
+    public Direction getDirection() {
+        return direction;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void warning() {
```

Comment:
same: please define methods after their usage

Reply:
Already change order of methods in the classes.